source/posix/thread.c

```diff
@@ -101,6 +135,7 @@ int aws_thread_launch(
     wrapper->allocator = thread->allocator;
     wrapper->func = func;
     wrapper->arg = arg;
+    wrapper->atexit = NULL;
```

Comment:
Now that you're adding stuff, why not nuke this and switch the malloc to a calloc?

Reply:
ok